Source/Dafny/Resolver.cs

```diff
@@ -3344,9 +3357,9 @@ public class ModuleBindings
               }
 
               break;
-            // FIXME - check for out of range e1
-            // case BinaryExpr.Opcode.LeftShift:  return BigInteger.LeftShift(e0, (int)e1);
-            // case BinaryExpr.Opcode.RightShift: return BigInteger.RightShift(e0, (int)e1);
+            // FIXME - check these for sign of e1; do bit trimming?
+            case BinaryExpr.Opcode.LeftShift:  return (BigInteger)e0 << (int)e1;
+            case BinaryExpr.Opcode.RightShift: return (BigInteger)e0 >> (int)e1;
```

Comment:
I'd like these fixed before we merge, so that we check all preconditions before the constant folding takes place.

Also, is the cast `(int)e1` really correct? I would have expect `e1` to be a `BigInteger` at this time.

Reply:
Fixed and tests written